File: 2020/python/day1.py
#!/usr/bin/env python3
import sys


def day1(input_file):
    with open(input_file, "r") as f:
        input = f.read()

    print("Day 1: Report Repair")
    print(f"\tPart 1: {__part1(input)}")
    print(f"\tPart 2: {__part2(input)}")


def __part1(input):
    input = [int(x) for x in input.splitlines()]

    for i in range(len(input)):
        for j in range(i + 1, len(input)):
            if input[i] + input[j] == 2020:
                return input[i] * input[j]


def __part2(input):
    input = [int(x) for x in input.splitlines()]

    for i in range(len(input)):
        for j in range(i + 1, len(input)):
            for k in range(j + 1, len(input)):
                if input[i] + input[j] + input[k] == 2020:
                    return input[i] * input[j] * input[k]


if __name__ == "__main__":
    day1(sys.argv[1])

File: 2020/javascript/day1.js
#!/usr/bin/env node

import fs from "fs";

const part1 = (input) => {
  const entries = input.split("\n").map((e) => parseInt(e));
  for (let i = 0; i < entries.length; i++) {
    for (let j = i + 1; j < entries.length; j++) {
      if (entries[i] + entries[j] === 2020) {
        return entries[i] * entries[j];
      }
    }
  }
};

const part2 = (input) => {
  const entries = input.split("\n").map((e) => parseInt(e));
  for (let i = 0; i < entries.length; i++) {
    for (let j = i + 1; j < entries.length; j++) {
      for (let k = j + 1; k < entries.length; k++) {
        if (entries[i] + entries[j] + entries[k] === 2020) {
          return entries[i] * entries[j] * entries[k];
        }
      }
    }
  }
};

const day1 = (inputFile) => {
  const input = fs.readFileSync(inputFile, "utf8");

  console.log("Day 1: Report Repair");
  console.log(`\tPart 1: ${part1(input)}`);
  console.log(`\tPart 2: ${part2(input)}`);
};

export default day1;

try {
  day1(process.argv[2]);
} catch (e) {
  console.log(`${e}`);
}
